pkg/client/client.go

```diff
@@ -246,20 +207,212 @@ func Initialize(
 	}
 }
 
+func checkAwaitingKeyGeneration(
+	ctx context.Context,
+	ethereumChain eth.Handle,
+	tssNode *node.Node,
+	operatorPublicKey *operator.PublicKey,
+	keepsRegistry *registry.Keeps,
+	requestedSignatures *requestedSignaturesTrack,
+) {
+	keepCount, err := ethereumChain.GetKeepCount()
+	if err != nil {
+		logger.Warningf("could not get keep count: [%v]", err)
+		return
+	}
+
+	zero := big.NewInt(0)
+	one := big.NewInt(1)
+
+	lastIndex := new(big.Int).Sub(keepCount, one)
+
+	// Iterate through keeps starting from the end.
+	for keepIndex := new(big.Int).Set(lastIndex); keepIndex.Cmp(zero) != -1; keepIndex.Sub(keepIndex, one) {
+
```

Comment:
Can we add a debug log that we are checking keep at the given index for key generation?